webapp/packages/plugin-connection-template/src/ConnectionDialog/ConnectionDialog.tsx

```diff
@@ -17,82 +17,89 @@ import {
   const styles = useS(style);
-  const [focusedRef] = useFocus<HTMLFormElement>({ focusFirstChild: true });
-  const controller = useController(ConnectionController, rejectDialog);
   const translate = useTranslate();
+  const [focusedRef] = useFocus<HTMLFormElement>({ focusFirstChild: true });
+  const dbDriverResource = useResource(ConnectionDialog, DBDriverResource, CachedMapAllKey);
   const { credentialsSavingEnabled } = useAdministrationSettings();
+  const dialog = useConnectionDialog(rejectDialog);
+  const errorDetails = useErrorDetails(dialog.connectException);
 
-  let subtitle: string | undefined;
+  useAutoLoad(ConnectionDialog, dialog);
 
-  if (controller.step === ConnectionStep.Connection && controller.template?.name) {
-    subtitle = controller.template.name;
-  }
+  const subTitle = dialog.step === ConnectionStep.Connection ? dialog.template?.name : undefined;
 
   return (
-    <CommonDialogWrapper size="large" fixedSize>
+    <CommonDialogWrapper size="large" fixedSize={dialog.step === ConnectionStep.ConnectionTemplateSelect}>
       <CommonDialogHeader
         title="basicConnection_connectionDialog_newConnection"
-        subTitle={subtitle}
-        icon={controller.dbDriver?.icon}
+        subTitle={subTitle}
+        icon={dialog.driver?.icon}
         onReject={rejectDialog}
       />
-      <CommonDialogBody noBodyPadding={controller.step === ConnectionStep.ConnectionTemplateSelect} noOverflow>
-        {controller.isLoading && <Loader />}
-        {!controller.isLoading && controller.step === ConnectionStep.ConnectionTemplateSelect && (
-          <TemplateConnectionSelector
-            templateConnections={controller.templateConnections}
-            dbDrivers={controller.dbDrivers}
-            onSelect={controller.onTemplateSelect}
-          />
-        )}
-        {controller.step === ConnectionStep.Connection &&
-          (!controller.authModel ? (
-            <center className={s(styles, { center: true })}>
-              {controller.isConnecting && translate('basicConnection_connectionDialog_connecting_message')}
-            </center>
-          ) : (
-            <Form ref={focusedRef} className={s(styles, { submittingForm: true })} onSubmit={controller.onConnect}>
-              <ConnectionAuthenticationFormLoader
-                config={controller.config}
-                authModelId={controller.authModel.id}
-                networkHandlers={controller.networkHandlers}
-                formId={controller.template?.id}
-                allowSaveCredentials={credentialsSavingEnabled}
-                disabled={controller.isConnecting}
-                className={s(styles, { connectionAuthenticationFormLoader: true })}
-              />
-            </Form>
-          ))}
+      <CommonDialogBody noBodyPadding={dialog.step === ConnectionStep.ConnectionTemplateSelect} noOverflow>
+        <Loader state={[dialog, dbDriverResource]}>
```

Comment:
please remove loader and use suspense (maybe you will need to add `<Loader suspense>` here but it's already exist in dialog component)